examples/expression-template-workflow.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: expression-template-
+  labels:
+    workflows.argoproj.io/test: "true"
+  annotations:
+    # available in v3.1.0
+    workflows.argoproj.io/version: ">= 3.1.0"
+    workflows.argoproj.io/verify.py: |
+      assert status["phase"] == "Succeeded"
+      assert nodes["task-0(0:3)"]["phase"] == "Succeeded"
+      assert nodes["task-0(0:3)"]["outputs"]["parameters"][0]["value"] == "hello 30 @ 2021"
+spec:
+  entrypoint: main
+  templates:
+    - name: main
+      dag:
+        tasks:
+          - name: task-0
+            template: pod-0
+            arguments:
+              parameters:
+                - name: foo
+                  value: "{{=item}}"
+            # withParam must be a JSON list encoded as a string, so you use `toJson` to perform the conversion
+            withParam: "{{=sprig.toJson(filter([1, 3], {# > 1}))}}"
```

Comment:
Since this is an example, it might be useful to link to https://github.com/Masterminds/sprig as reference to additional syntax for people not familiar with it

Reply:
better docs overall I think